### DISCUSSION TITLE
Change NextApiRequest query value to string instead of string | string[]

### DISCUSSION THREAD

**Discussion by wyeo** [vercel/next.js#21594] [Ideas]
**Describe the feature you'd like to request**

File : `next-server/lib/utils.d.ts`
Replace NextApiRequest.query string | string [] by just string

**Describe the solution you'd like**
```
export interface NextApiRequest extends IncomingMessage {
    /**
     * Object of `query` values from url
     */
    query: {
        [key: string]: string;
    };
    ...
}
```

**Describe alternatives you've considered**

- Use an API Middleware to format the **string** received from the request
- Format the **string** value directly in the handler function like **req.query.userIds.split(',')**
- Overriding the req type as `{ NextApiRequest & query: { [ key: string]: string } }`

**Comment by awareness481** [comment]
`String[]` is a valid type as you can have `?query=one,two` and so on

**Reply by wyeo** [reply]
With`?query=one,two` **query** is a string and not string[]
